app/Http/Controllers/Bank/NordigenController.php

```diff
@@ -17,270 +17,254 @@
-            ]);
+            return $this->failed('requisition-not-found', $context, $company);
         }
-        if ($requisition["status"] != "LN") {
-            return view('bank.nordigen.handler', [
-                'lang' => $lang,
-                'company' => $company,
-                'account' => $company->account,
-                'failed_reason' => "requisition-invalid-status",
-                "redirectUrl" => $context["redirect"] . "?action=nordigen_connect&status=failed&reason=requisition-invalid-status&status=" . $requisition["status"],
-            ]);
+        if ($requisition['status'] != 'LN') {
+            return $this->failed('requisition-invalid-status&status=' . $requisition['status'], $context, $company);
         }
-        if (sizeof($requisition["accounts"]) == 0) {
-            return view('bank.nordigen.handler', [
-                'lang' => $lang,
-                'company' => $company,
-                'account' => $company->account,
-                'failed_reason' => "requisition-no-accounts",
-                "redirectUrl" => $context["redirect"] . "?action=nordigen_connect&status=failed&reason=requisition-no-accounts",
-            ]);
+        if (sizeof($requisition['accounts']) == 0) {
+            return $this->failed('requisition-no-accounts', $context, $company);
         }
 
         // connect new accounts
         $bank_integration_ids = [];
-        foreach ($requisition["accounts"] as $nordigenAccountId) {
-
+        foreach ($requisition['accounts'] as $nordigenAccountId) {
             $nordigen_account = $nordigen->getAccount($nordigenAccountId);
 
             if (isset($nordigen_account['error'])) {
                 continue;
             }
 
-            $existing_bank_integration = BankIntegration::withTrashed()->where('nordigen_account_id', $nordigen_account['id'])->where('company_id', $company->id)->where('is_deleted', 0)->first();
-
-            if (!$existing_bank_integration) {
+            try {
+                $bank_integration = $this->findIntegrationBy('account', $nordigen_account, $company);
 
+                $bank_integration->deleted_at = null;
+            } catch (ModelNotFoundException $e) {
                 $bank_integration = new BankIntegration();
+
                 $bank_integration->integration_type = BankIntegration::INTEGRATION_TYPE_NORDIGEN;
                 $bank_integration->company_id = $company->id;
                 $bank_integration->account_id = $company->account_id;
                 $bank_integration->user_id = $company->owner()->id;
                 $bank_integration->nordigen_account_id = $nordigen_account['id'];
                 $bank_integration->bank_account_type = $nordigen_account['account_type'];
                 $bank_integration->bank_account_name = $nordigen_account['account_name'];
-                $bank_integration->bank_account_status = $nordigen_account['account_status'];
```

Comment:
please explain this change :)

Reply:
From f37edfdabaf501c3cbb3fbfaa77350d93a5954ae - Instead of an if/else block where some things (like status and balance) are duplicated, the BankIntegration query now throws an exception so the non-unique bits are applied once in the `finally` instead.